include/nccl_ofi_mr.h

```diff
@@ -204,6 +219,21 @@ int nccl_ofi_mr_cache_insert_entry(nccl_ofi_mr_cache_t *cache, nccl_ofi_mr_ckey_
  */
 int nccl_ofi_mr_cache_del_entry(nccl_ofi_mr_cache_t *cache, void *handle);
 
+/**
+ * Generate a duplicated key with start and len aligned to cache's
+ * system_page_size. This is for callers of nccl_ofi_mr_cache_insert_entry()
```

Comment:
That we need this function is a sign our APIs are wrong.  Let's fix that, rather than create this function.

Reply:
The below layer like libfabric takes a ckey as an input for actual memory registration. So it looks I have to generate another key at the same layer in the callers of nccl_ofi_mr_cache_insert_entry(). The logic of creating a copy of ckey (including address / length computation) with aligned base / bound are like what I have in this API. 

If I get the logic correctly, should we keep these logic as a single API? (If we have these logic in the caller, the code can be messy.)

Reply:
No, the underlying interfaces do not take a ckey as input.  They take a base and bound.  The ckey is entirely a construct we created.  There's no reason we need to dupe them to pad them out.  No reason not to do it at ckey creation, actually.  Many ways of solving this problem, but this one is hacky and error prone.

Reply:
Moved the aligned registration to the ckey creation.

Reply:
> No, the underlying interfaces do not take a ckey as input.

The first member of the ckey is always safe to cast to fi_mr_attr